addons/delivery/models/stock_picking.py

```diff
@@ -30,14 +30,12 @@ def _compute_weight(self):
             package.weight = weight
 
     def _get_default_weight_uom(self):
-        return self.env['product.template']._get_weight_uom_name_from_ir_config_parameter()
-
-    def _compute_weight_uom_name(self):
-        for package in self:
-            package.weight_uom_name = self.env['product.template']._get_weight_uom_name_from_ir_config_parameter()
+        return self.env['product.template']._get_weight_uom_id_from_ir_config_parameter()
 
     weight = fields.Float(compute='_compute_weight', help="Total weight of all the products contained in the package.")
-    weight_uom_name = fields.Char(string='Weight unit of measure label', compute='_compute_weight_uom_name', readonly=True, default=_get_default_weight_uom)
+    weight_uom_id = fields.Many2one(
```

Comment:
```suggestion
    weight_uom_name = fields.Char(related="weight_uom_id.display_name")
    weight_uom_id = fields.Many2one(
```
For the sake of retrocompatibility :wink: